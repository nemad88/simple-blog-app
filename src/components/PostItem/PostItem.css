.post-item {
    background: chocolate;
    font-size: 16px;
}