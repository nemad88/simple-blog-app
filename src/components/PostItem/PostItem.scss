.post-list-item {
    padding: 1rem 1rem 5rem 1rem;
    background: #D9D9D9;
    margin-bottom: 1.5rem;
    -webkit-box-shadow: 9px 10px 20px 0px rgba(0,0,0,0.75);
    -moz-box-shadow: 9px 10px 20px 0px rgba(0,0,0,0.75);
    box-shadow: 9px 10px 20px 0px rgba(0,0,0,0.75);


    h1 {
        color: #0D1821;
        padding: 0 1rem 0 0;

        text-transform: uppercase;
        width: 30%;
        font-size: 3rem;
        margin: 1rem 0;
        border-bottom: 15px solid #0D1821;

    }

    p{
        font-size: 1.8rem;
        margin: 3rem 0;
    }

    a {
        color: black;
        text-decoration: none;
        display: inline-block;
        font-size: 1.4rem;
        border: 1px solid #0D1821;
        background: #0D1821;
        color: #D9D9D9;
        padding: 1rem;
        -webkit-box-shadow: 9px 10px 20px -8px rgba(0,0,0,0.75);
        -moz-box-shadow: 9px 10px 20px -8px rgba(0,0,0,0.75);
        box-shadow: 9px 10px 20px -8px rgba(0,0,0,0.75);
        margin-right: 1rem;
        text-align: center;
        width: 140px;

        &:hover {
            background: #D9D9D9;
            color: #0D1821;
            border: 1px solid #0D1821;
        }

    }

}