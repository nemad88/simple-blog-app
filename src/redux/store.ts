import {combineReducers, createStore, applyMiddleware} from 'redux';
import thunk from "redux-thunk";
import postsReducer from './posts';
import commentReducer from './comments';

const rootReducer = combineReducers({
    posts: postsReducer,
    comments: commentReducer
})

export type RootState = ReturnType<typeof rootReducer>;

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;

