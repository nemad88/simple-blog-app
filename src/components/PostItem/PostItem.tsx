import React from "react";
import {Post} from "../../redux/posts";

interface Props {
    post: Post;
}

const PostItem: React.FC<Props> = ({post}) => {
    return (
        <div>
            <h2> {post.title}</h2>
            <p>{post.body}</p>
        </div>
    )
}

export default PostItem;