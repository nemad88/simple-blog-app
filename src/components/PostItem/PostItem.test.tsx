import React from "react";
import { shallow } from "enzyme";
import PostItem from "./PostItem";
import { Provider } from "react-redux";
import store from "../../redux/store";

it("expect to render PostItem component", () => {
  expect(
    shallow(
      <Provider store={store}>
        <PostItem
          post={{
            body:
              "et fugit eligendi deleniti quidem qui sint nihil autem m accusamus fugiat dicta voluptatem rerum ut voluptate autem voluptatem repellendus aspernatur dolorem in doloribus at sed quis culpa deserunt consectetur qui praesen\ndoloribus at sed quis culpa deserunt consectetur qui praesentium accusamus fugiat dicta voluptatem rerum ut voluptate autem voluptatem repellendus aspernatur dolorem in\ndoloribus at sed quis culpa deserunt uidem qui sint nihil autem m accusamus fugiat dicta voluptatem rerum ut voluptate autem voluptatem repellendus aspernatur dolorem in doloribus at sed quis culpa deserunt consectetur qui praesen consectetur qui praesen praesentium accusamus fugiat dicta voluptatem rerum ut voluptate autem voluptatem repellendus aspernatur dolorem in\ndoloribus at sed quis culpa deserunt consectetur qui praesen",
            visibility: "draft",
            id: 2,
            title: "qui est esse",
            userId: 1,
          }}
        />
      </Provider>
    )
  ).toMatchSnapshot();
});
