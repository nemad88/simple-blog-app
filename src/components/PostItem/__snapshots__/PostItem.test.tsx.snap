// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render PostItem component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <PostItem
    post={
      Object {
        "body": "et fugit eligendi deleniti quidem qui sint nihil autem m accusamus fugiat dicta voluptatem rerum ut voluptate autem voluptatem repellendus aspernatur dolorem in doloribus at sed quis culpa deserunt consectetur qui praesen
doloribus at sed quis culpa deserunt consectetur qui praesentium accusamus fugiat dicta voluptatem rerum ut voluptate autem voluptatem repellendus aspernatur dolorem in
doloribus at sed quis culpa deserunt uidem qui sint nihil autem m accusamus fugiat dicta voluptatem rerum ut voluptate autem voluptatem repellendus aspernatur dolorem in doloribus at sed quis culpa deserunt consectetur qui praesen consectetur qui praesen praesentium accusamus fugiat dicta voluptatem rerum ut voluptate autem voluptatem repellendus aspernatur dolorem in
doloribus at sed quis culpa deserunt consectetur qui praesen",
        "id": 2,
        "title": "qui est esse",
        "userId": 1,
        "visibility": "draft",
      }
    }
  />
</ContextProvider>
`;
